src/main/java/io/cdap/e2e/pages/actions/CdfPipelineRunAction.java

```diff
@@ -127,15 +127,38 @@ public static void waitForPipelineToTransitionToStatus(String pipelineStatus, lo
    * Timeout: {@link ConstantsUtil#IMPLICIT_TIMEOUT_SECONDS}
    */
   public static void waitTillPipelineRunCompletes() throws InterruptedException {
-    // Adding a page refresh in case tests are running on CDF to update the pipeline status.
-      RetryUtils.retry(ConstantsUtil.PIPELINE_REFRESH_TIMEOUT_SECONDS, ConstantsUtil.PIPELINE_RUN_TIMEOUT_SECONDS,
-        10, () -> {
+    try {
+      // Wait for the Pipeline run to complete for the default timeout
+      WaitHelper.waitForElementToBeHidden(CdfPipelineRunLocators.runningStatus);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+
+
+    // Loop to refresh the page if the pipeline is still running till max refresh count
+    int maxRefreshCount = ConstantsUtil.MAX_PAGE_REFRESH_ATTEMPTS;
+    while (maxRefreshCount > 0) {
+      boolean isPipelineRunning = isRunning();
+
+      if (isPipelineRunning) {
         PageHelper.refreshCurrentPage();
-        return !(isRunning());
+
+        try {
+          Thread.sleep(5000); // Wait for 5 second after the refresh
```

Comment:
Had to use the static wait as there are multiple threads running in parallel and UI responsiveness can introduce unwanted failures

Reply:
can we move this to constant class?